docs/rl-algorithms/sac.md

```diff
@@ -0,0 +1,224 @@
+            log_std = self.fc_logstd(x)
+            log_std = torch.tanh(log_std)
+            log_std = LOG_STD_MIN + 0.5 * (LOG_STD_MAX - LOG_STD_MIN) * (log_std + 1)  # From SpinUp / Denis Yarats
+
+            return mean, log_std
+
+        def get_action(self, x):
+            mean, log_std = self(x)
+            std = log_std.exp()
+            normal = torch.distributions.Normal(mean, std)
+            x_t = normal.rsample()  # for reparameterization trick (mean + std * N(0,1))
+            y_t = torch.tanh(x_t)
+            action = y_t * self.action_scale + self.action_bias
+            log_prob = normal.log_prob(x_t)
+            # Enforcing Action Bound
+            log_prob -= torch.log(self.action_scale * (1 - y_t.pow(2)) + 1e-6)
+            log_prob = log_prob.sum(1, keepdim=True)
+            mean = torch.tanh(mean) * self.action_scale + self.action_bias
+            return action, log_prob, mean
+
+        def to(self, device):
+            self.action_scale = self.action_scale.to(device)
+            self.action_bias = self.action_bias.to(device)
+            return super(Actor, self).to(device)
+    ```
+    Note that unlike :material-github: [openai/spinningup](https://github.com/openai/spinningup/tree/master/spinup/algos/tf1/sac)'s implementation which uses `LOG_STD_MIN = -20`, CleanRL's uses `LOG_STD_MIN = -5` instead.
+
+2. [`sac_continuous_action.py`](https://github.com/vwxyzjn/cleanrl/blob/master/cleanrl/sac_continuous_action.py) uses different learning rates for the policy and the Soft Q-value networks optimization.
+
+    ```python
+        parser.add_argument("--policy-lr", type=float, default=3e-4,
+            help="the learning rate of the policy network optimizer")
+        parser.add_argument("--q-lr", type=float, default=1e-3,
+            help="the learning rate of the Q network network optimizer")
+    ```
+    while [openai/spinningup](https://github.com/openai/spinningup/blob/038665d62d569055401d91856abb287263096178/spinup/algos/tf1/sac/sac.py#L44)'s uses a single learning rate of `lr=1e-3` for both components.
+
+    Note that in case it is used, the *automatic entropy coefficient* $\alpha$'s tuning shares the `q-lr` learning rate:
+    ```python hl_lines="6"
+        # Automatic entropy tuning
+        if args.autotune:
+            target_entropy = -torch.prod(torch.Tensor(envs.single_action_space.shape).to(device)).item()
+            log_alpha = torch.zeros(1, requires_grad=True, device=device)
+            alpha = log_alpha.exp().item()
+            a_optimizer = optim.Adam([log_alpha], lr=args.q_lr)
+        else:
+            alpha = args.alpha
+    ```
+
+3. [`sac_continuous_action.py`](https://github.com/vwxyzjn/cleanrl/blob/master/cleanrl/sac_continuous_action.py) uses `--batch-size=256` while :material-github: [openai/spinningup](https://github.com/openai/spinningup/blob/038665d62d569055401d91856abb287263096178/spinup/algos/tf1/sac/sac.py#L44)'s uses `--batch-size=100` by default.
+
```

Comment:
I'd also mention that we have used a global gradient clipping which is an optimization detail.

Reply:
Done, but maybe too extensively ?